lib/constants.js

```diff
@@ -36,7 +36,9 @@ var Constants = {
         AAD_API_VERSION : 'api-version',
         USERNAME : 'username',
         PASSWORD : 'password',
-        REFRESH_TOKEN : 'refresh_token'
+        REFRESH_TOKEN : 'refresh_token', 
+        LANGUAGE : 'mkt', 
+        DEVICE_CODE : 'device_code', 
```

Comment:
I might be looking at the wrong specs, but where did you get `device_code` as a grant type from? This document says it should be `device_token`, but it also says the field name should be `type` instead of `grant_type` so I'm a little confused.

https://tools.ietf.org/html/draft-ietf-oauth-v2-05#section-3.7.2